src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php

```diff
@@ -267,7 +267,10 @@ private function parseDefinition(\DOMElement $service, $file, array $defaults)
         foreach (array('class', 'public', 'shared', 'synthetic', 'lazy', 'abstract') as $key) {
             if ($value = $service->getAttribute($key)) {
                 $method = 'set'.$key;
-                $definition->$method(XmlUtils::phpize($value));
+                $definition->$method($value = XmlUtils::phpize($value));
+                if ('lazy' === $key && \is_string($value)) {
```

Comment:
as `lazy` has a special logic, I would rather remove it from the loop and handle it on its own (as done for some other properties)